### PR TITLE
[cda] Fix and improve extraction

### LINKED ISSUE #10929: CDA plugin won't work anymore
As in title, they changed website structure...

### PR DESCRIPTION
**Pull request by kasper93**
### Before submitting a _pull request_ make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)
### What is the purpose of your _pull request_?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

Fixes #10929

### CODE REVIEW

File: youtube_dl/extractor/cda.py [hunk 1]
```diff
@@ -38,32 +48,71 @@ class CDAIE(InfoExtractor):
...
+        view_count = self._search_regex(
+            r'Odsłony:(?:\s|&nbsp;)*([0-9]+)', webpage, 'view_count', default=None)
+
+        m_rating = re.search(RATING_REGEX, webpage)
+        average_rating = None
+        if m_rating:
+            rating_value = float_or_none(m_rating.group('rating_value'))
+            best_rating = float_or_none(m_rating.group('best_rating'))
+            if rating_value and best_rating:
+                average_rating = rating_value / best_rating
```
Comment (kasper93) [inline youtube_dl/extractor/cda.py]: I wasn't sure if `average_rating` need to be scaled to 1 or not. If not we can simplify this to just `rating_value`

Comment (dstftw) [inline youtube_dl/extractor/cda.py]: It should not. Read the description of the field.

File: youtube_dl/extractor/cda.py [hunk 2]
```diff
@@ -38,32 +48,73 @@ class CDAIE(InfoExtractor):
...
+                expected=True)
 
         formats = []
 
+        # Set uploader to None if not found.
+        # There are many videos without uploader specified.
+        uploader = self._search_regex(UPLOADER_REGEX, webpage, 'uploader',
+                                      default=None, group='uploader')
+
+        # Set view_count to None if not found.
```
Comment (dstftw) [inline youtube_dl/extractor/cda.py]: All pointless comments should be removed.

File: youtube_dl/extractor/cda.py (hunk 1 above)
Comment (kasper93) [inline youtube_dl/extractor/cda.py]: Done.

File: youtube_dl/extractor/cda.py (hunk 2 above)
Comment (kasper93) [inline youtube_dl/extractor/cda.py]: Done.

File: youtube_dl/extractor/cda.py [hunk 3]
```diff
@@ -38,32 +48,56 @@ class CDAIE(InfoExtractor):
...
 
     def _real_extract(self, url):
+        UPLOADER_REGEX = r'''(?x)
+            <(span|meta)[^>]*itemprop=(["\'])author\2[^>]*>
+            (?:<\1[^>]*>[^<]*</\1>|(?!</\1>)(?:.|\n))*?
+            <(span|meta)[^>]*itemprop=(["\'])name\4[^>]*>(?P<uploader>[^<]+)</\3>
+        '''
+        RATING_REGEX = r'<(?:span|meta)[^>]*itemprop=(["\'])ratingValue\1[^>]*>(?P<rating_value>[0-9.]+)'
+        VIEWS_REGEX = r'Odsłony:(?:\s|&nbsp;)*([0-9]+)'
+        PDATA_REGEX = r'player_data=(\\?["\'])(?P<player_data>.+?)\1'
```
Comment (dstftw) [inline youtube_dl/extractor/cda.py]: All these regexes are only used once thus make no sense as separate variables.

File: youtube_dl/extractor/cda.py [hunk 4]
```diff
@@ -75,21 +109,21 @@ def extract_format(page, version):
...
         extract_format(webpage, 'default')
 
         for href, resolution in re.findall(
                 r'<a[^>]+data-quality="[^"]+"[^>]+href="([^"]+)"[^>]+class="quality-btn"[^>]*>([0-9]+p)',
                 webpage):
-            webpage = self._download_webpage(
-                href, video_id, 'Downloading %s version information' % resolution, fatal=False)
+            webpage = self._download_webpage(self._BASE_URL + href, video_id,
+                                             'Downloading %s version information'
+                                             % resolution, fatal=False)
```
Comment (dstftw) [inline youtube_dl/extractor/cda.py]: What's the point changing the original indent? With this style you just waste lots of space.

File: youtube_dl/extractor/cda.py [hunk 5]
```diff
@@ -39,31 +49,49 @@ class CDAIE(InfoExtractor):
...
 
         def extract_format(page, version):
-            unpacked = decode_packed_codes(page)
-            format_url = self._search_regex(
-                r"(?:file|url)\s*:\s*(\\?[\"'])(?P<url>http.+?)\1", unpacked,
-                '%s url' % version, fatal=False, group='url')
-            if not format_url:
+            player_data = self._parse_json(self._search_regex(
+                r'player_data=(\\?["\'])(?P<player_data>.+?)\1', page,
+                '%s player_data' % version, fatal=False, group='player_data'), video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/cda.py]: If `_search_regex` fails `None` will be passed to `_parse_json`.

**Comment by dstftw** [comment]
Cherry-picked, thanks.

**Pull request closed without merging**